### PR TITLE
Allow to set base href on compilation time

### PR DESCRIPTION
**Pull request by alagane**
Related to linagora/esn-frontend-common-libs#217

### CODE REVIEW

File: webpack.commons.js [hunk 1]
```diff
@@ -247,13 +247,14 @@ module.exports = {
...
-    contentBasePublicPath: [BASE_HREF, '/login'],
-    publicPath: '/mailto/',
+    contentBase: [path.resolve(__dirname, 'node_modules', 'esn-frontend-login', 'dist')],
+    contentBasePublicPath: ['/login'],
     compress: true,
     host: '0.0.0.0',
     disableHostCheck: true,
-    historyApiFallback: true,
+    historyApiFallback: {
+      index: BASE_HREF + 'index.html'
```
Comment (tanandy) [inline webpack.commons.js:256]: are we sure about this  and hardcoded index.html ?  (just a question)

Comment (alagane) [reply webpack.commons.js:256]: This in only in dev server mode, and I think it won't change, and if it does we can change the hardcoded value.

**Comment by alagane** [comment]
I did not get the project to work correctly, just to show the default page (the app banner with the app grid...), I think it is because the generate does not authenticate.

**Comment by asmaNode** [comment]
what kind of error you have , because i had the some problem in the other SPA (did not work )

**Comment by alagane** [comment]
# Mailto SPA

Configuration used:
```
COMPOSE_FILE=docker-compose.yml:docker-compose.esn.yml:./preview/docker-compose.esn.preview.yml:docker-compose.mailto.yml
```
Image linagora/esn-frontend-mailto:latest is built locally on commit https://github.com/linagora/esn-frontend-mailto/commit/6ad6d8368885b3b9b44f2a00430eb51cd3923eee (was main latest commit)
Seems to be the same behaviour with this pull request: https://github.com/linagora/esn-frontend-mailto/pull/78

## Failed requests

```
GET /api/user?_=1623321060660 HTTP/1.1
Host: backend.open-paas.org.local
User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:88.0) Gecko/20100101 Firefox/88.0
Accept: application/json, text/plain, */*
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate
Origin: http://mailto.open-paas.org.local
Connection: keep-alive
Referer: http://mailto.open-paas.org.local/
```
Got 401 (without Authorization header, without cookie)
If tried with Authorization header with a valid token, works correctly (200)

```
POST /api/jwt/generate HTTP/1.1
Host: backend.open-paas.org.local
User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:88.0) Gecko/20100101 Firefox/88.0
Accept: application/json, text/plain, */*
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate
Content-Type: application/json;charset=utf-8
Content-Length: 2
Origin: http://mailto.open-paas.org.local
Connection: keep-alive
Referer: http://mailto.open-paas.org.local/
Cache-Control: max-age=0
```
Got 401 (without Authorization header, without cookie)
If tried with Authorization header with a valid token, works correctly (200)

```
GET /api/themes/60c085f1f2b848008264025a/logo HTTP/1.1
Host: backend.open-paas.org.local
User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:88.0) Gecko/20100101 Firefox/88.0
Accept: image/webp,*/*
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate
Connection: keep-alive
Referer: http://mailto.open-paas.org.local/
Cache-Control: max-age=0
```
Got 401 (without Authorization header, without cookie)
If tried with Authorization header with a valid token, responds with a 401

## http://auth.open-paas.org.local/auth/realms/master/protocol/openid-connect/token

Is called successfully 2 times on connection, but one time should be enough
Get access_token (given on connection) we can request the user API (and the generate if the token is still valid)

### Example

```
eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJWZE4yWXgzZEdBckw5eDNzVUFzSzY3V0NKcnpLMWF5MGZWUUEzR3Z0Qzg4In0.eyJleHAiOjE2MjMzMjg0NTYsImlhdCI6MTYyMzMyNDg1NiwiYXV0aF90aW1lIjoxNjIzMzI0ODUzLCJqdGkiOiIyODJkOWJmYy00YWI0LTQwOTAtYTQ3NS1kMGZkZDc4OTg2NWYiLCJpc3MiOiJodHRwOi8vYXV0aC5vcGVuLXBhYXMub3JnLmxvY2FsL2F1dGgvcmVhbG1zL21hc3RlciIsImF1ZCI6ImFjY291bnQiLCJzdWIiOiI2MDk5YTFmOS1lNjIyLTQ2MTItYmI5OS1lM2NjZGZhZGI2YTIiLCJ0eXAiOiJCZWFyZXIiLCJhenAiOiJvcGVucGFhcyIsInNlc3Npb25fc3RhdGUiOiJkNmVmZDlmNi1jNzY2LTRlN2ItOTYyYS1lZjgyMDdmOTQ5NzciLCJhY3IiOiIxIiwiYWxsb3dlZC1vcmlnaW5zIjpbIioiXSwicmVhbG1fYWNjZXNzIjp7InJvbGVzIjpbImRlZmF1bHQtcm9sZXMtbWFzdGVyIiwib2ZmbGluZV9hY2Nlc3MiLCJ1bWFfYXV0aG9yaXphdGlvbiJdfSwicmVzb3VyY2VfYWNjZXNzIjp7ImFjY291bnQiOnsicm9sZXMiOlsibWFuYWdlLWFjY291bnQiLCJtYW5hZ2UtYWNjb3VudC1saW5rcyIsInZpZXctcHJvZmlsZSJdfX0sInNjb3BlIjoib3BlbmlkIGVtYWlsIHByb2ZpbGUiLCJlbWFpbF92ZXJpZmllZCI6ZmFsc2UsIm5hbWUiOiJBbGV4YW5kcmUgTGFnYW5lIiwicHJlZmVycmVkX3VzZXJuYW1lIjoiYWxhZ2FuZSIsImdpdmVuX25hbWUiOiJBbGV4YW5kcmUiLCJmYW1pbHlfbmFtZSI6IkxhZ2FuZSIsImVtYWlsIjoiYWxhZ2FuZUBvcGVuLXBhYXMub3JnIn0.K1Ey-KZCmr8W09zdLWwTPrLBD3GwvrUhNQ-_OgvQUm6ecNucx8lmdg0aHw7D4AzheBYEw7rx0SSEL8lGFcBWqUztZdbzQ2u7Tqwkj3GZPJ8ji0SjFqYy4DnpQZR4pGu6sqQ4TJ1v16o_cigm4586V0he4eZr9NJ_flgG0QipislAMEARFvgqN6DIzPBHcXrrbntYJM194TB_w1-emdaGUWkD4HbOwOzqbVpCBCAqepXIhn-j333xoakp_gOxCtxTibEb_TJ5iihd-m2XAOJZ0z2mtIdNuf6-kELm2pqH3G4UoczCNloMogeRj4RhWbP_Dd81KejO1qj3CFeIZGRXxA
```
This one is correctly signed with the public key found here: http://auth.open-paas.org.local/auth/realms/master

## http://backend.open-paas.org.local/api/jwt/generate

Generate gives a new access token if a provided token is valid, the new new token is refreshed
It gives much less informations than the initial token, but both can be used

### Example

```
eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJhbGFnYW5lQG9wZW4tcGFhcy5vcmciLCJpYXQiOjE2MjMzMjUxNzEsImV4cCI6MTYyMzM2ODM3MX0.X2Jo7OPTthlkSdj-yRh3O-VXh0F1LOIkGb4yIUZU8pMrukmt_54ml_MNVpxxaiO_lz5propg51pb8yo2LMn_Nky-iZAf49hcY1kGNfP1G7uBAEadsKd29sQxefKJtpu6ogP1DZJImVh0xEeVW-6_tQvOjf53WUo1y4GdrSFy11Y01PJDc1FZfvUrfeSXGACJrpDJn5E7SDiuY8rR0jzjjaMt7Nw_1tzDIS2gWaXFv2Y0DqwTOuVL_kDLwLHm4e9-jlxVOYoBTd7gLQhi6Sspkl6L61IzmSrLoPyjfRzlmqie-r1P1MpSzaLUgvkiHcYOnBaNG6PbRYlcyDKr0Ip8vw
```
This one is correctly signed with the public key found in MongoDB configuration

If refreshing the page and then using cookies or session :
- Cookies do not seem to work, token seems not valied
- Session seems to work, using the access token

**Review by tanandy — APPROVED** [review]

**Review by oubchid — APPROVED** [review]

**Pull request merged**